packages/web3-core-method/src/index.js

```diff
@@ -770,19 +771,15 @@ Method.prototype.buildCall = function () {
         };
 
         // Send the actual transaction
-        if (isSendTx && !!payload.params[0] && typeof payload.params[0] === 'object' && typeof payload.params[0].gasPrice === 'undefined') {
+        if (isSendTx 
+            && !!payload.params[0]
+            && typeof payload.params[0] === 'object'
+        ) {
+            if (typeof payload.params[0].type === 'undefined') 
+                payload.params[0].type = _handleTxType(payload.params[0]);
 
-            var getGasPrice = (new Method({
-                name: 'getGasPrice',
-                call: 'eth_gasPrice',
-                params: 0
-            })).createFunction(method.requestManager);
-
-            getGasPrice(function (err, gasPrice) {
-
-                if (gasPrice) {
-                    payload.params[0].gasPrice = gasPrice;
-                }
+            _handleTxPricing(method, payload.params[0]).then(txPricing => {
+                payload.params[0] = {...payload.params[0], ...txPricing};
```

Comment:
Would it make more sense to move all this logic outside of the blocks?

Regardless of **how** its being sent, we should just update the TX accordingly

Reply:
I think in this case, we only want to add gas pricing info to the `payload` object if the info is not already provided for specifically [sendTxs](https://github.com/ChainSafe/web3.js/blob/b5dadf9ef2352a1378c26937d03d4109fd3319b0/packages/web3-core-method/src/index.js#L612), otherwise the code shouldn't be executed